src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/ContainerExecDecorator.java

```diff
@@ -118,18 +123,31 @@ public Proc launch(ProcStarter starter) throws IOException {
                 boolean quiet = starter.quiet();
                 FilePath pwd = starter.pwd();
 
-                String [] cmdEnvs = starter.envs();
 
-                //check if the cmd is sourced from Jenkins, rather than another plugin; if so, skip cmdEnvs as we are getting other environment variables
-                for (String cmd : cmdEnvs) {
-                    if (cmd.startsWith(JENKINS_HOME)) {
-                        cmdEnvs = new String[0];
-                        LOGGER.info("Skipping injection of procstarter cmdenvs due to JENKINS_HOME present");
+                List<String> procStarter =  Arrays.asList(starter.envs());
+                List<String> cmdEnvs = new  ArrayList<String>();
+                //check if the cmd is sourced from Jenkins, rather than another plugin; if so, skip cmdEnvs except for built-in ones.
+                boolean javaHome_detected = false;
+                for(String env: procStarter)
+                {
+                    if(env.contains("JAVA_HOME"))
```

Comment:
I don't understand what does it mean if `JAVA_HOME` is there, what does that mean?

Reply:
and due to the `break` any variable after `JAVA_HOME` is ignored, so the order of vars matter